jrnl/upgrade.py

```diff
@@ -85,17 +86,27 @@ def upgrade_jrnl_if_necessary(config_path):
         util.prompt("\nUpgrading encrypted '{}' journal stored in {}...".format(journal_name, path))
         backup(path, binary=True)
         old_journal = Journal.open_journal(journal_name, util.scope_config(config, journal_name), legacy=True)
-        new_journal = EncryptedJournal.from_journal(old_journal)
-        new_journal.write()
-        util.prompt("  Done.")
+        all_journals.append(EncryptedJournal.from_journal(old_journal))
 
     for journal_name, path in plain_journals.items():
         util.prompt("\nUpgrading plain text '{}' journal stored in {}...".format(journal_name, path))
         backup(path)
         old_journal = Journal.open_journal(journal_name, util.scope_config(config, journal_name), legacy=True)
-        new_journal = Journal.PlainJournal.from_journal(old_journal)
-        new_journal.write()
-        util.prompt("  Done.")
+        all_journals.append(Journal.PlainJournal.from_journal(old_journal))
+
+    # loop through lists to validate
+    failed_journals = [j for j in all_journals if not j.validate_parsing()]
+    if len(failed_journals) > 0:
+        util.prompt("\nThe following journal{} failed to upgrade:\n{}".format(
+            's' if len(failed_journals) > 1 else '', "\n".join(j.name for j in failed_journals))
+        )
+
+        util.prompt("Aborting upgrade.")
+        return
```

Comment:
We should use an exception here instead of returning. Once this function returns, a bunch of other code will still run (like upgrading the config), and not fully abort the upgrade.

Reply:
Agreed - just made the change.